pkg/pss/pss.go

```diff
@@ -116,18 +112,24 @@ func (p *pss) Register(topic trojan.Topic, handler Handler) (cleanup func()) {
 	}
 }
 
+func (p *pss) topics() (ts []Topic) {
+	p.handlersMu.Lock()
+	defer p.handlersMu.Unlock()
+	for t, _ := range p.handlers {
+		ts = append(ts, t)
```

Comment:
This is just something that can be easily done, but probably without so significant performance gain, `ts` slice can be preallocated with `make([]Topic, 0, len(p.handlers))` before the  for loop.